### PR TITLE
Remove extra none in the legend of 1D plots

### LINKED ISSUE #28090: Additional 'none' item in the legend of 1D plots
If the legend of a 1D plot is modified on Ubuntu an extra 'none' entry will appear in the legend. 

Its possible the legend needs to be redrawn using ax.legend() followed by fig.canvas.draw(). For instance, the following steps remove the extra entry. 

(1) Plot the data and add errors, and then go to the IPython window and do:

`fig = plt.gcf()`
`ax = plt.gca()`
`ax.legend()`
`fig.canvas.draw()`

Edit: This also occurs in fitting
image

### Steps to reproduce the behavior
(1) Load some data
(2) Plot a spectra
(3) Add errors
(4) An extra none entry will appear in the legend 

### Platforms affected
Appears to be limited to workbench on Ubuntu.

### PR DESCRIPTION
**Pull request by StephenSmith25**
**Description of work.**
If the legend of a 1D plot was modified (by re-plotting data with errors for example) on Ubuntu an extra 'none' entry appeared in the legend. This was due to older versions of Matplotlib casting None to a string, see the commit https://github.com/matplotlib/matplotlib/commit/84292c46902913273d9231414e19bd11664117a6

This PR adds in a check before the legend is generated, to remove the 'None' string.

**To test:**

1. Open workbench on Ubuntu ( but presumably other distributions (and possibly mac?)
2. Plot any data
3. Right-click and add errors
4. A spurious 'None' entry should no longer appear. 

Fixes #28090 . 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by cailafinn — CHANGES_REQUESTED** [review]
Changes function as expected. Just a comment on making the code neater.

File: Framework/PythonInterface/mantid/plots/legend.py [hunk 1]
```diff
@@ -127,6 +127,13 @@ def from_view_advanced(cls, view):
 
     @classmethod
     def create_legend(cls, props, ax):
+        # For Matplotlib <3.2 we have to remove the 'None' string from the title
+        # which is generated by set_text() in text.py
+        title = props['title']
+        if title == 'None':
+            props['title'] = ''
+        else:
+            props['title'] = title
```
[the remark this answers is not in the document]
Comment (StephenSmith25) [reply Framework/PythonInterface/mantid/plots/legend.py]: > 
> 
> I'm not sure this else is necessary, if the title is already `title`.

Yeah good point, not sure what I was thinking there.

**Review by cailafinn — APPROVED** [review]
Still functions as described, code looks good. Nice work. :+1:

**Comment by martyngigg** [comment]
Unrelated build failure.

File: Framework/PythonInterface/mantid/plots/legend.py [hunk 2]
```diff
@@ -127,6 +127,10 @@ def from_view_advanced(cls, view):
 
     @classmethod
     def create_legend(cls, props, ax):
+        # For Matplotlib <3.2 we have to remove the 'None' string from the title
```
Comment (martyngigg) [inline Framework/PythonInterface/mantid/plots/legend.py]: @StephenSmith25 I meant to ask. Would it be possible to add unit test somewhere for this?

Comment (StephenSmith25) [reply Framework/PythonInterface/mantid/plots/legend.py]: Yeah, I can write a test somewhere for it. Should I create a separate PR for this?

Comment (martyngigg) [reply Framework/PythonInterface/mantid/plots/legend.py]: :+1: Thanks. Yeah please create a separate PR.

**Pull request merged**